java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -6494,12 +6511,66 @@ void testGroupByContiguousSplitGroups() {
         splits = table.groupBy().contiguousSplitGroups();
         assertEquals(1, splits.length);
         assertTablesAreEqual(table, splits[0].getTable());
+
+        GroupByResult r = table.groupBy().contiguousSplitGroupsAndGenUniqKeys();
+        splits2 = r.getGroups();
+        assertEquals(1, splits2.length);
+        assertTablesAreEqual(table, splits2[0].getTable());
+        uniqKeys = r.getUniqKeyTable();
+
+        // Table should contain 1 or more columns,
+        // If group by empty, keys table should be null;
+        assertNull(uniqKeys);
       } finally {
         if (splits != null) {
           for (ContiguousTable t : splits) { t.close(); }
         }
+        if(splits2 != null) {
+          for (ContiguousTable t : splits2) { t.close(); }
+        }
+        if(uniqKeys != null) {
+          uniqKeys.close();
+        }
       }
 
+      // Row count is 0
+      ContiguousTable[] tmpSplits = null;
+      try {
+        // the first of tmpSplits is empty split
+        tmpSplits = table.contiguousSplit(0);
+        Table emptyTable = tmpSplits[0].getTable();
+        assertEquals(0, emptyTable.getRowCount());
+
+        splits = emptyTable.groupBy(0, 1).contiguousSplitGroups();
+        assertEquals(1, splits.length);
+        assertEquals(0, splits[0].getTable().getRowCount());
+
+        GroupByResult r = emptyTable.groupBy(0, 1).contiguousSplitGroupsAndGenUniqKeys();
+        splits2 = r.getGroups();
+        assertEquals(1, splits2.length);
+        assertEquals(0, splits2[0].getTable().getRowCount());
+        uniqKeys = r.getUniqKeyTable();
+        assertEquals(0, uniqKeys.getRowCount());
+      } finally {
+        if (tmpSplits != null) {
+          for (ContiguousTable t : tmpSplits) {
+            t.close();
+          }
+        }
+        if (splits != null) {
+          for (ContiguousTable t : splits) {
+            t.close();
+          }
+        }
+        if (splits2 != null) {
+          for (ContiguousTable t : splits2) {
+            t.close();
+          }
+        }
+        if (uniqKeys != null) {
+          uniqKeys.close();
+        }
```

Comment:
Why don't we have auto close? Manually closing resources is bad IMO.

Reply:
Done